Sources/OpenAI/Private/JSONRequest.swift

```diff
@@ -33,7 +33,9 @@ extension JSONRequest: URLRequestBuildable {
             request.setValue(organizationIdentifier, forHTTPHeaderField: "OpenAI-Organization")
         }
         request.httpMethod = method
-        request.httpBody = try JSONEncoder().encode(body)
+        if method != "GET" && method != "DELETE" {
```

Comment:
What do you think of extracting this condition into a function or variable? Something like `requiresBody: Bool`?

Thanks for contributing, really appreciate it 👍

Reply:
SGTM 😄 
I've updated the PR to use the optional body instead.
On the other side, `ModelsQuery` itself is not necessary too. Do you want to keep it?

Reply:
Yes, looks like `ModelsQuery` is unused. Do you mind removing it with this PR? Along with `models` function update. Looks like YAGNI.

Reply:
No problem. Added a commit to remove `ModelsQuery` with updating the `models()` function.